src/analysis/retail/demonhunter/vengeance/normalizers/ShearFractureNormalizer.ts

```diff
@@ -0,0 +1,65 @@
+import SPELLS from 'common/SPELLS/demonhunter';
+import TALENTS from 'common/TALENTS/demonhunter';
+import { CastEvent, EventType, GetRelatedEvents, RemoveBuffStackEvent } from 'parser/core/Events';
+import { Options } from 'parser/core/Module';
+import EventLinkNormalizer, { EventLink } from 'parser/core/EventLinkNormalizer';
+
+/*
+WCL: https://www.warcraftlogs.com/reports/Y7BWyCx3mHVZzPrk#fight=last&type=summary&view=events&pins=2%24Off%24%23244F4B%24casts%7Cauras%24-1%240.0.0.Any%240.0.0.Any%24true%240.0.0.Any%24true%24258920%7C204255%7C203981%7C263642
+Default spell is Shear (generates 1 soul). Fracture (generates 2 souls) talent replaces it.
+Vengence can have a maxium of 5 souls. Log has 4 fracture casts. Meaning 5 gained souls and 3 wasted. 2 of the 4 casts are bad.
+Souls are generated on a slight delay. Souls should be cast within 1.3 seconds.
+If a soul is cast and then a buff is removed within 100 ms that means it was a wasted soul generation.
+ */
+
+// Note that this number may need to change with sufficient levels of haste.
+const SOUL_GENERATE_BUFFER = 1300;
+const SOUL_WASTED_BUFFER = 100;
+
+const GENERATED_SOUL_FRAGMENT = 'ShearFractureGeneratedSoulFragment';
+const WASTED_SOUL_FRAGMENT = 'ShearFractureWastedSoulFragment';
+
+const EVENT_LINKS: EventLink[] = [
+  {
+    linkRelation: GENERATED_SOUL_FRAGMENT,
+    referencedEventId: [SPELLS.SHEAR.id, TALENTS.FRACTURE_VENGEANCE_TALENT.id],
+    referencedEventType: EventType.Cast,
+    linkingEventId: SPELLS.SOUL_FRAGMENT.id,
+    linkingEventType: EventType.Cast,
+    forwardBufferMs: SOUL_GENERATE_BUFFER,
+    backwardBufferMs: SOUL_GENERATE_BUFFER,
```

Comment:
If my reading of the behavior is correct, the long delay always comes *after* the Shear / Fracture cast - does this work with forward buffer of zero, but backwards buffer the same? Am I misunderstanding what's going on?

Reply:
Setting it to a `forwardBufferMs` of `0` shows as having 64 bad casts, which is more accurate. I'm still trying to run down another false positive in one of the logs I'm working with, but this in itself should be enough.